rpm/qtlocation.spec

```diff
@@ -162,7 +162,11 @@ find %{buildroot}%{_libdir} -type f -name '*.prl' \
 %post -n qt5-qtpositioning -p /sbin/ldconfig
 %postun -n qt5-qtpositioning -p /sbin/ldconfig
 
-%post -n qt5-qtlocation -p /sbin/ldconfig
+%post -n qt5-qtlocation
+/sbin/ldconfig
+# this directory contains a bunch of cached images, potentially with the wrong tile size.
+/bin/rm -rf /home/nemo/.cache/org.sailfishos/sailfish-maps || :
```

Comment:
This could perhaps be made a bit smarter and only remove map tiles which are not 512x512. Otherwise it complicates the package upgrade process. In the next package update the above would need to be removed so that valid 512x512 tiles are not unnecessarily removed. But then if the user happens to skips the initial update they will hit the original bug that this change is fixing.

Also this might not be the best package for this change. /home/nemo/.cache/org.sailfishos/sailfish-maps is the cache directory used exclusively by sailfish-maps, /home/nemo/.cache/QtLocation (the default cache directory) and other cache directories from other applications are not cleared. Though there are no such applications on Jolla devices delivered through the Jolla store.

A more generic solution may be to reject the loaded tiles if the size does not match the tile size set in the manager. This way the tile will be immediately evicted (well replaced) in the cache when first accessed.